test/server/lib/DocApi.ts

```diff
@@ -3180,6 +3180,122 @@ function testDocApi() {
       accessResp = await axios.patch(`${homeUrl}/api/docs/${docIds.Timesheets}/access`, {delta}, chimpy);
       assert.equal(accessResp.status, 200);
     });
+
+    it ("GET /docs/{did}/sql works", async function () {
+      const resp = await axios.get(`${serverUrl}/api/docs/${docIds.Timesheets}/sql?q=select+*+from+Table1+order+by+id`, chimpy);
+      assert.equal(resp.status, 200);
+      assert.deepEqual(resp.data, {
+        statement: 'select * from Table1 order by id',
+        records: [
+          {
+            fields: {
+              id: 1,
+              manualSort: 1,
+              A: 'hello',
+              B: '',
+              C: '',
+              D: null,
+              E: 'HELLO'
+            },
+          },
+          {
+            fields: { id: 2, manualSort: 2, A: '', B: 'world', C: '', D: null, E: '' }
+          },
+          {
+            fields: { id: 3, manualSort: 3, A: '', B: '', C: '', D: null, E: '' }
+          },
+          {
+            fields: { id: 4, manualSort: 4, A: '', B: '', C: '', D: null, E: '' }
+          },
+        ]
+      });
+    });
+
+    it ("POST /docs/{did}/sql works", async function () {
+      let resp = await axios.post(
+        `${serverUrl}/api/docs/${docIds.Timesheets}/sql`,
+        { sql: "select A from Table1 where id = ?", args: [ 1 ] },
+        chimpy);
+      assert.equal(resp.status, 200);
+      assert.deepEqual(resp.data.records, [{
+        fields: {
+          A: 'hello',
+        }
+      }]);
+
+      resp = await axios.post(
+        `${serverUrl}/api/docs/${docIds.Timesheets}/sql`,
+        { nosql: "select A from Table1 where id = ?", args: [ 1 ] },
+        chimpy);
+      assert.equal(resp.status, 400);
+      assert.deepEqual(resp.data, {
+        error: 'Invalid payload',
+        details: { userError: 'Error: value.sql is missing' }
+      });
+
+      resp = await axios.post(
+        `${serverUrl}/api/docs/${docIds.Timesheets}/sql`,
+        { sql: "select A from Table1 where id = ?", args: [ 1 ] },
+        nobody);
+      assert.equal(resp.status, 403);
+      assert.deepEqual(resp.data, {
+        error: 'No view access',
+      });
```

Comment:
I know it's a draft, sorry if I bother you already with my comments :grinning: (this endpoints looks promising and raised my attention! :eyes:)

Do you think it could be doable to add a test to check that a user having partial access to the document gets a 403? (if you don't already have in mind to test that)